cargo/private/cargo_build_script_wrapper.bzl

```diff
@@ -112,7 +112,7 @@ def cargo_build_script(
         binary_tags.append("manual")
     build_script_kwargs = {}
     if "compatible_with" in kwargs:
```

Comment:
hm, this is fine logically but it's a bit of mental gymnastics to figure out what's happening.
For readability, could we maybe introduce a new variable `binary_kwargs`, initialized with kwargs and pop the `compatible_with` from that one, leaving the `kwargs` unchanged?
Also, could we update the docs `**kwargs: Forwards to the underlying `rust_binary` rule.` a bit to spell out the compatible_with behavior?

Reply:
Done.